test/recipes/05-test_sm4.t

```diff
@@ -0,0 +1,12 @@
+#! /usr/bin/env perl
+# Copyright 2015-2016 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the OpenSSL license (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+
+use OpenSSL::Test::Simple;
+
+simple_test("test_internal_sm4", "sm4_internal_test", "sm4");
```

Comment:
Please rename this file to go along with the other internal tests.  `03-test_internal_sm4.t` would be a good one.

Also, since this is only built under certain condicions, you need to add the following as well, before the `simple_test` line:

``` perl
plan skip_all => "This test is unsupported in a shared library build on Windows"
    if $^O eq 'MSWin32' && !disabled("shared");
```

Reply:
Done, thanks @levitte !